benchmarks/01_trivial/04_fastdiv.cpp

```diff
@@ -0,0 +1,113 @@
+#include <benchmark/benchmark.h>
+#include <bitset>
+#include <cstdint>
+#include <stdio.h>
+
+#include "../common.h"
+#include "04_fastdiv_ispc.h"
+
+static Docs docs("Check fast_idiv implmentation of stdlib functions:\n"
+                 "[int8, int16, int32] x [2, 13, 16, 127] x [-ve, +ve] versions.\n"
```

Comment:
- you don't need different power of 2, they trigger the same code.
- is int64 triggering fastdiv? I think it makes sense to have it to see it's perfomance, even if it doesn't trigger fastdiv.
- positive and negative means that algorithm is covering positive and negative vlaues. I.e. ``signed int`` should trigger algorithm covering both negative and positive, while ``unsigned int`` should trigger algorithm, which covers only positives. I'm not sure that range analysis takes care about it and is used in fastdiv.

Reply:
1. I meant 16 as power of 2, 13 as non-power of 2 and 2 and 127 as outer bounds of values. I have no logical reasoning for thinking the outer bounds will behave any differently but decided to have those anyway. I have taken out 2 in the update
2. int64 does not trigger fast_idiv. But have added it for comparison.
3 Not sure about this. My intention was to just test the algorithm with negative values . I have changed to use uint value. Our earlier implementation for this seem to be a bit of a mess - https://godbolt.org/z/fDfuJf  : see warning

Reply:
- comments were not changed
- 13 and 127 triggered code. which looked different, but if you don't have reasoning leave only 13.
- you should not have two different functions for testing positive and negative values. You should either pass these values or produce them internally without a negation operator. I think if you pass them in another array (say src[]), that would be better. Otherwise it creates unnecessary complications and confuses.
- README still needs an update.

Reply:
1. Comments changed
2. Done
3. This more complicated. Adding a src[] makes it messier in my opinion. The numerator needs to be a varying  variable and cannot be an array so src[i]/divVal doesn't work. Using the src[] is some other way means adding a gather too. I have tweaked it to make it a little more cleaner.
4. Updated..its missing 03_popcnt too.

Reply:
1-2. ok.
3. ``src[i]/divVal`` works perfectly fine, as ``src[i]`` is varying value in case of varying ``i``. Avoiding extra perations in the loops makes it cleaner test for division. Even though range arithmetic will likely be negligible, but it's better to avoid it. memory load from L1 should be very fast, especially as HW can overlap different iterations. 
4. Yes, I'll add popcnt description.

Reply:
If you add ``src[]``, you'll be able to handle sign and unsigned version in uniform way.

Reply:
3. Fair enough. I have changed the ispc side.